### PR TITLE
Documentation for PR #797 in demo app -  Collector Data Flow Dashboard

### PR DESCRIPTION
**Pull request by devrimdemiroz**
Documentation for https://github.com/open-telemetry/opentelemetry-demo/pull/797 on opentelemetry demo application repo.

---

**Preview**: https://deploy-preview-2538--opentelemetry.netlify.app/docs/demo/collector-data-flow-dashboard/

### CODE REVIEW

**Review by cartermp — CHANGES_REQUESTED** [review]
Overall LGTM, could you use code fences for the names of the metrics listed here?

**Comment by devrimdemiroz** [comment]
> Overall LGTM, could you use code fences for the names of the metrics listed here?

`fenced`

**Review by cartermp — APPROVED** [review]

File: content/en/docs/demo/collector-data-flow-dashboard.md
Comment (cartermp) [inline content/en/docs/demo/collector-data-flow-dashboard.md:13]:
```suggestion
![](/docs/demo/screenshots/otelcol-data-flow-overview.png)
```

File: content/en/docs/demo/collector-data-flow-dashboard.md
Comment (cartermp) [inline content/en/docs/demo/collector-data-flow-dashboard.md:72]:
```suggestion
![](/docs/demo/screenshots/otelcol-data-flow-dashboard.png)
```

File: content/en/docs/demo/collector-data-flow-dashboard.md
Comment (cartermp) [inline content/en/docs/demo/collector-data-flow-dashboard.md:99]:
```suggestion
![](/docs/demo/screenshots/otelcol-dashbord-process-metrics.png)
```

File: content/en/docs/demo/collector-data-flow-dashboard.md
Comment (cartermp) [inline content/en/docs/demo/collector-data-flow-dashboard.md:22]:
```suggestion
![](/docs/demo/screenshots/otelcol-data-flow-metrics.png) The metrics depicted in the
```

**Review by cartermp — COMMENTED** [review]
This should fix the link issues.

File: content/en/docs/demo/collector-data-flow-dashboard.md
Comment (devrimdemiroz) [reply content/en/docs/demo/collector-data-flow-dashboard.md]: I was trying understand what was the link issue. @cartermp , thanks!

**Review by chalin — CHANGES_REQUESTED** [review]
Currently the newly added page doesn't render (so we can't preview it) because it has no Hugo front matter. I can suggest some front-matter content later if you'd like, but my first question is, **where is the new page meant to fit in the overall demo docs**? IMHO, we don't want it at the top level.

@open-telemetry/demo-approvers have any comment about this?

The newly created images shouldn't be in the `screenshots` folder unless they show up in the **Screenshots** page.

File: content/en/docs/demo/collector-data-flow-dashboard.md
Comment (chalin) [inline content/en/docs/demo/collector-data-flow-dashboard.md:15]: All images (here and elsewhere in the document) should have (nonempty) `alt` text (inside the square brackets).

File: content/en/docs/demo/collector-data-flow-dashboard.md
Comment (chalin) [inline content/en/docs/demo/collector-data-flow-dashboard.md:25]:
```suggestion
![](/docs/demo/screenshots/otelcol-data-flow-metrics.png)

The metrics depicted in the
diagram above are employed to monitor both egress and ingress data flows. These
```

File: content/en/docs/demo/collector-data-flow-dashboard.md
Comment (chalin) [inline content/en/docs/demo/collector-data-flow-dashboard.md:6]: macro-level --> high-level?

File: content/en/docs/demo/collector-data-flow-dashboard.md
Comment (chalin) [inline content/en/docs/demo/collector-data-flow-dashboard.md:11]: It isn't obvious to me how you're "offering a solid foundation for users to build upon".

File: content/en/docs/demo/collector-data-flow-dashboard.md
Comment (chalin) [inline content/en/docs/demo/collector-data-flow-dashboard.md:13]: This section isn't an overview of the dashboard, right? That's the impression we get. The section title needs adjustment iMHO.

Comment (devrimdemiroz) [reply content/en/docs/demo/collector-data-flow-dashboard.md:13]: You are right, more of a system overview. So in a sense, not just the dashboard overview. May be "o11y Data Flow Overview"... Any other suggestion?

File: content/en/docs/demo/collector-data-flow-dashboard.md
Comment (devrimdemiroz) [reply content/en/docs/demo/collector-data-flow-dashboard.md:6]: I used macro-level on purpose as later when you have many collectors in place, the board can provide macro-level visibility. As per the demo, I agree, it is just high-level. Shall I change?

**Comment by devrimdemiroz** [comment]
> Currently the newly added page doesn't render (so we can't preview it) because it has no Hugo front matter. I can suggest some front-matter content later if you'd like, but my first question is, **where is the new page meant to fit in the overall demo docs**? IMHO, we don't want it at the top level.
> 
> @open-telemetry/demo-approvers have any comment about this?
> 
> The newly created images shouldn't be in the `screenshots` folder unless they show up in the **Screenshots** page.

I did not find a better place inside this directory. I am happy to receive a recommendation on this one.

**Comment by chalin** [comment]
> I did not find a better place inside this directory.

No worries, that's fine. Waiting on @open-telemetry/demo-approvers and others to weigh in.

File: content/en/docs/demo/collector-data-flow-dashboard.md
Comment (devrimdemiroz) [reply content/en/docs/demo/collector-data-flow-dashboard.md]: `... offering a solid foundation for users to build upon. Collector Data Flow Dashboard provides valuable guidance on which metrics to monitor. Users can tailor their own dashboard variations by adding necessary metrics specific to their use cases, such as memory_delimiter processor or other o11y data flow indicators. This demo dashboard serves as a starting point, enabling users to explore diverse usage scenarios and adapt the tool to their unique monitoring needs.`

? 🤓

File: content/en/docs/demo/collector-data-flow-dashboard.md
Comment (julianocosta89) [inline content/en/docs/demo/collector-data-flow-dashboard.md:14]: Do we use `o11y` somewhere else in the docs?
Even though it's a known abbreviation, I'd rather have it as observability here, just to keep it simple.

**Review by julianocosta89 — APPROVED** [review]
Just a comment, but not a blocker.
LGTM

Thanks for taking care of that @devrimdemiroz

File: content/en/docs/demo/collector-data-flow-dashboard.md
Comment (devrimdemiroz) [reply content/en/docs/demo/collector-data-flow-dashboard.md]: Thanks for noticing such detail. It felt strange when I put it in the first place.  I agree, I removed it now.👌🏽

**Review by svrnm — APPROVED** [review]
lgtm! Great addition

**Comment by julianocosta89** [comment]
> Currently the newly added page doesn't render (so we can't preview it) because it has no Hugo front matter. I can suggest some front-matter content later if you'd like, but my first question is, **where is the new page meant to fit in the overall demo docs**? IMHO, we don't want it at the top level.
> 
> @open-telemetry/demo-approvers have any comment about this?
> 
> The newly created images shouldn't be in the `screenshots` folder unless they show up in the **Screenshots** page.

**where is the new page meant to fit in the overall demo docs**?
@cartersocha and @puckpuck any ideas on that?

**Comment by cartersocha** [comment]
Glancing through the current TOC. I think it’d be nice to have a dashboard folder or something similar. We could do similar write ups for the other pre existing dashboards?

**Comment by chalin** [comment]
> Glancing through the current TOC. I think it’d be nice to have a dashboard folder or something similar. We could do similar write ups for the other pre existing dashboards?

Thanks all for your feedback. 

If you feel that other dashboards will be documented soonish, then I'd vote to create **Dashboards** section with the single page in the PR as an entry. Otherwise, we can leave the page as it is (at the top level) and move it later as other dashboards are added. Your call @cartersocha et al.

File: content/en/docs/demo/collector-data-flow-dashboard.md
Comment (chalin) [inline content/en/docs/demo/collector-data-flow-dashboard.md:2]: Would "Data Flow Dashboard" make sense as a side-nav title?

```suggestion
title: Collector Data Flow Dashboard
linkTitle: Data Flow Dashboard
```

**Review by chalin — CHANGES_REQUESTED** [review]
@devrimdemiroz given that the page will stay at the top level for now, you'll need to make the following changes:

- Create a folder named `collector-data-flow-dashboard`
- Rename `content/en/docs/demo/collector-data-flow-dashboard.md` to `content/en/docs/demo/collector-data-flow-dashboard/index.md`
- Move all of the images of this PR into `content/en/docs/demo/collector-data-flow-dashboard`
- In `index.md`, adjust the path to the images by dropping everything except the file name.

File: content/en/docs/demo/collector-data-flow-dashboard.md
Comment (chalin) [reply content/en/docs/demo/collector-data-flow-dashboard.md:20]: Marking as unresolved since I'm not seeing any alt text.

File: content/en/docs/demo/collector-data-flow-dashboard.md
Comment (chalin) [inline content/en/docs/demo/collector-data-flow-dashboard.md:81]: Add alt text here.

File: content/en/docs/demo/collector-data-flow-dashboard.md
Comment (chalin) [reply content/en/docs/demo/collector-data-flow-dashboard.md:29]: Move the image after the section's opening paragraph.

File: content/en/docs/demo/collector-data-flow-dashboard.md
Comment (chalin) [inline content/en/docs/demo/collector-data-flow-dashboard.md:31]:
```suggestion
The metrics depicted in the diagram below are employed to monitor both egress
```

File: content/en/docs/demo/collector-data-flow-dashboard.md
Comment (chalin) [reply content/en/docs/demo/collector-data-flow-dashboard.md:20]: Move the image after the opening paragraph.

File: content/en/docs/demo/collector-data-flow-dashboard.md
Comment (chalin) [inline content/en/docs/demo/collector-data-flow-dashboard.md:22]:
```suggestion
The diagram below provides an overview of the system components, showcasing the
```

File: content/en/docs/demo/collector-data-flow-dashboard.md
Comment (devrimdemiroz) [reply content/en/docs/demo/collector-data-flow-dashboard.md:29]: I need to understand, these ones were already corrected. I am looking into it.

File: content/en/docs/demo/collector-data-flow-dashboard.md
Comment (devrimdemiroz) [reply content/en/docs/demo/collector-data-flow-dashboard.md]: added

**Review by chalin — COMMENTED** [review]
Thanks for the updates @devrimdemiroz. We're getting closer!

I've unresolved some of my comments since they haven't been addressed. Please avoid marking a comment as resolved unless the required changes have landed in the PR. If there are questions or doubts, leave it unresolved so that the comment author or maintainers can make the assessment and update the status as required or follow up with more comments.

The main issue that's left is _image placement_. We _generally_ want to avoid starting a section with an image. In all cases in the page, the opening paragraph is short enough that the image can be placed _after_ the opening paragraph. Please make that change for all images, and adjust the paragraph text to refer to the "image below" rather than "above.

I'll have a few final copyedits to share after that. Thanks!

File: content/en/docs/demo/collector-data-flow-dashboard/index.md
Comment (chalin) [inline content/en/docs/demo/collector-data-flow-dashboard/index.md:58]:
```suggestion
Currently, the only processor present in the demo application is a batch
processor, which is used by both traces and metrics pipelines.
```

File: content/en/docs/demo/collector-data-flow-dashboard/index.md
Comment (chalin) [inline content/en/docs/demo/collector-data-flow-dashboard/index.md:87]:
```suggestion
The dashboard has four main sections:
```

File: content/en/docs/demo/collector-data-flow-dashboard.md
Comment (cartermp) [reply content/en/docs/demo/collector-data-flow-dashboard.md]: Works fro me.

**Comment by devrimdemiroz** [comment]
> Thanks for the updates @devrimdemiroz. We're getting closer!
> 
> I've unresolved some of my comments since they haven't been addressed. Please avoid marking a comment as resolved unless the required changes have landed in the PR. If there are questions or doubts, leave it unresolved so that the comment author or maintainers can make the assessment and update the status as required or follow up with more comments.
> 
> The main issue that's left is _image placement_. We _generally_ want to avoid starting a section with an image. In all cases in the page, the opening paragraph is short enough that the image can be placed _after_ the opening paragraph. Please make that change for all images, and adjust the paragraph text to refer to the "image below" rather than "above.
> 
> I'll have a few final copyedits to share after that. Thanks!

@chalin ,

I appreciate the thorough review you've given to the document. In my case, I believe that placing the diagram first offers some distinct advantages:

Context: Providing the diagram upfront supplies readers with a visual context before delving into the explanation, simplifying their comprehension of the content.

Engagement: Introducing a visual aid at the beginning captures readers' attention and piques their curiosity, increasing the likelihood that they'll engage with the subsequent text.

I hope you can understand my perspective on this. If it's not a strict requirement, I'd prefer to maintain the current format. While I recognize that there's no one-size-fits-all rule, my choice to place the diagrams first was deliberate. By investing in high-quality visuals, I aim to ensure that the related text is not overlooked and that the reader benefits from the full context.

Would you consider being open to this approach, as it may lead to a more engaging and effective document for our readers? I truly believe that this layout has the potential to enhance their overall experience.

Looking forward to your thoughts.
PS: Not limited but my favourite mentor on the subject 👉🏾 https://youtu.be/VwocXCyOieM

**Comment by chalin** [comment]
Thanks for the explanation @devrimdemiroz.

I agree with the **advantages** you are stating (engaging visuals etc) and IMHO, these **advantages _aren't lost_** by placing the images 2-3 lines further down the page. 

One of the hats that I wear for this project is that of technical writer. In that role, I do my best to **assure some form of consistency** in the website content presentation, reasonably enforcing "best / recommended practices" that I've accumulated over the years. (Like trying to avoid "widowed or orphaned" headings.)

I'll let @svrnm @cartermp and @austinlparker share their thoughts, but for sizeable open-source projects like OTel, consistency in style and formatting across content (blog post are given a bit more flexibility), trumps personal style/preference, otherwise we'd end up with a degraded reader experience due to the use of disparate styles. Consistency is also why, for example, we've enforced the use of Prettier to format file content. 

The only other solution I see, to keep the images first, would be to place the image in a figure _before_ the heading, and then refer to the figure by number or title.

Cheers!

**Comment by austinlparker** [comment]
I'd agree with @chalin -- we should strive for consistency.

**Comment by devrimdemiroz** [comment]
`The Individual CLA option is not enabled for this project. Please to your administrator to enable the Individual CLA option for this project.`

I wonder why individual CLA is not enabled 🧐

**Comment by chalin** [comment]
Hi @devrimdemiroz. Thanks for making the updates.

The CLA issue seems to be because you submitted the last commit from a different user ID, for which you haven't signed the CLA:

```console
$ g log -2  
commit 121862d7923790ade0d3628a7d24cf43a229f31f (HEAD -> feature/otecol-data-flow-dashboard-PR797)
Author: Devrim <devrim.demiroz@swisscom.com>
Date:   Fri Mar 31 09:21:34 2023 +0200

    high above towards down below

commit c7a72d123618b886052ca43ab330f8581ffb04ae
Merge: f3d9ed5b 146de686
Author: Devrim Demiroz <62701796+devrimdemiroz@users.noreply.github.com>
Date:   Fri Mar 31 09:09:57 2023 +0200

    Merge branch 'open-telemetry:main' into feature/otecol-data-flow-dashboard-PR797
```

I see two way's forward. Sign the CLA for the ID used in https://github.com/open-telemetry/opentelemetry.io/commit/121862d7923790ade0d3628a7d24cf43a229f31f, or retract the commit and resubmit it using the same ID you did all the other work with.

**Comment by chalin** [comment]
> I do not recall such an error on opentelemetry-demo repository.

I just checked the demo repo, and there you never used your `...+devrimdemiroz@users...` ID.

**Comment by devrimdemiroz** [comment]
In any case, I made a new dummy commit that should resolve the issue. But still, EasyCLA check is in Required state.

File: content/en/docs/demo/collector-data-flow-dashboard.md
Comment (cartermp) [reply content/en/docs/demo/collector-data-flow-dashboard.md]: I don't mind either way. It's easy enough to change later if we need to.

**Comment by cartermp** [comment]
@devrimdemiroz I don't think the dummy commit helps since EasyCLA appears to check every commit. You could probably fixup that commit with an interactive rebase or re-open a new PR with the current document state (and none of the preceding commits). Either way, this is good enough for me to merge in when the license check is dealt with. I'd prefer not to use admin privilege to bypass that.

**Comment by devrimdemiroz** [comment]
> @devrimdemiroz I don't think the dummy commit helps since EasyCLA appears to check every commit. You could probably fixup that commit with an interactive rebase or re-open a new PR with the current document state (and none of the preceding commits). Either way, this is good enough for me to merge in when the license check is dealt with. I'd prefer not to use admin privilege to bypass that.

I concluded same. I will go for opening a new PR and close this one. Thanks!

**Comment by devrimdemiroz** [comment]
Opened a new PR to bypass EasyCLA https://github.com/open-telemetry/opentelemetry.io/pull/2569 
Closing this one...

**Pull request closed without merging**
